fetch/range/blob.any.js

```diff
@@ -0,0 +1,100 @@
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=0-5,15-"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with multiple range values");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range end should be greater than range start"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=10-5"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with short range end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no start or end"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=-"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range with no start or end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no end"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=10-",
+    }
+  });
+  return promise.then(function(resp) {
+    assert_equals(resp.status, 206, "HTTP status is 206")
+    assert_equals(resp.type, "basic", "response type is basic");
+    assert_equals(resp.headers.get("Content-Type"), "text/plain", "Content-Type is " + resp.headers.get("Content-Type"));
+    assert_equals(resp.headers.get("Content-Length"), "7", "Content-Length is " + resp.headers.get("Content-Length"));
+    assert_equals(resp.headers.get("Content-Range"), "bytes 10-16/17", "Content-Range is " + resp.headers.get("Content-Range"));
+  });
+}, "Blob range with no end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no start"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=-10",
+    }
+  });
+  return promise.then(function(resp) {
+    assert_equals(resp.status, 206, "HTTP status is 206")
+    assert_equals(resp.type, "basic", "response type is basic");
+    assert_equals(resp.headers.get("Content-Type"), "text/plain", "Content-Type is " + resp.headers.get("Content-Type"));
+    assert_equals(resp.headers.get("Content-Length"), "10", "Content-Length is " + resp.headers.get("Content-Length"));
+    assert_equals(resp.headers.get("Content-Range"), "bytes 9-18/19", "Content-Range is " + resp.headers.get("Content-Range"));
+  });
+}, "Blob range with no start");
+
+promise_test(function(test) {
+  var blob = new Blob(["TODO(dlroertson): what do we do when there are multiple range headers"],
+                      { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=0-5",
+      "Range": "bytes=15-"
```

Comment:
Not sure what to do here either.

Reply:
This should be parsed identically to `bytes=0-5,bytes=15-`, which is a bit weird due to how `ranges-specifier` is defined. Fortunately for us we don't have to support values with a comma so we don't have to worry about it too much.

Reply:
It's a little worrisome that both Chrome and Safari support this case (according to your table). I guess that means they don't have their parser operate on all headers, but rather one-by-one and pick the first or last. I think we should file bugs on them.